### DISCUSSION TITLE
Any opinions on hyperparameter tuning?

### DISCUSSION THREAD

**Discussion by powerhorse1986** [MaartenGr/BERTopic#2449] [Ideas]
Hi, dear BERTopic fans,

Our group is planning a project that aims to fine-tune the hyperparameters of UMAP and HDBSCAN. However, there are too many hyperparameters. For UMAP, there are n_neighbors, n_components, min_dist, and metric. For HDBSCAN, we have min_cluster_size, min_samples, metric, and prediction_data. Do you have any recommended strategies other than the classic grid search? Thank you so much.

Li

**Comment by MaartenGr** [comment]
It really depends on your specific use case and what you attempt to achieve. I would advise trying out some of the hyperparameters first yourself to get a feeling of what works and what doesn't. Especially with something that is so use case specific, I would argue focusing on manual tweaking first and aligning that with the use case / stakeholder(s).
